File: ios/Plugin/CapacitorLottieSplashScreenPlugin.swift
import Foundation
import Capacitor

@objc(CapacitorLottieSplashScreenPlugin)
public class CapacitorLottieSplashScreenPlugin: CAPPlugin {
    public static var isEnabledStatic = true
    private let implementation = CapacitorLottieSplashScreen()
    
    
    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
    
    override public func load() {
        if(CapacitorLottieSplashScreenPlugin.isEnabledStatic){
            let isEnabled = getConfigValue("Enabled") as? Bool ?? true
            if(isEnabled) {
                implementation.loadLottie(view: self.bridge?.viewController?.view, path: self.getConfigValue("LottieAnimationLocation") as? String)
            }
            implementation.onAnimationEvent = onAnimationEvent
        }
    }
    
    public func onAnimationEvent(event: AnimationEventListener) {
        print("onAnimationEvent",event.listenerEvent)
        self.bridge?.triggerWindowJSEvent(eventName: event.listenerEvent)
        self.notifyListeners(event.listenerEvent, data: nil)
    }
    
    @objc func appLoaded(_ call: CAPPluginCall) {
        implementation.onAppLoaded()
        call.resolve()
    }
    
    @objc func isAnimating(_ call: CAPPluginCall) {
        call.resolve([
            "isAnimating": implementation.isAnimating()
        ])
    }
    
}

File: android/src/main/java/com/getlocal/plugins/caplss/CapacitorLottieSplashScreenPlugin.java
package com.getlocal.plugins.caplss;

import android.animation.Animator;
import android.app.Dialog;
import android.content.Context;
import android.content.Intent;
import android.graphics.Color;
import android.util.AttributeSet;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;
import android.widget.LinearLayout;

import androidx.appcompat.app.AppCompatActivity;

import com.airbnb.lottie.LottieAnimationView;
import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.PluginResult;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "CapacitorLottieSplashScreen")
public class CapacitorLottieSplashScreenPlugin extends Plugin {
    private final CapacitorLottieSplashScreen implementation = new CapacitorLottieSplashScreen();

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);
    }

    public void load() {
        boolean isEnabled = this.getConfig().getBoolean("Enabled", true);
        if (isEnabled) ShowLottieSplashScreenDialog();
        implementation.setAnimationEventListener(this::onAnimationEvent);
    }

    public void onAnimationEvent(String event) {
        bridge.triggerWindowJSEvent(event);
        notifyListeners(event, null);
    }

    private void ShowLottieSplashScreenDialog() {
        Context context = this.getContext();
        String lottiePath = this.getConfig().getString("LottieAnimationLocation");
        implementation.ShowLottieSplashScreenDialog(context, lottiePath);
    }

    @PluginMethod
    public void appLoaded(PluginCall call) {
        implementation.onAppLoaded();
        call.resolve();
    }

    @PluginMethod
    public void isAnimating(PluginCall call) {
        boolean isAnimating = implementation.isAnimating();
        JSObject ret = new JSObject();
        ret.put("isAnimating", isAnimating);
        call.resolve(ret);
    }
}
